### ISSUE TITLE
Does test butler work with android-library modules?

### ISSUE THREAD

**Issue by zsperske** [linkedin/test-butler#83]
I'm writing an android-library module and test butler fails to grant read/write permissions on a stock API 25 emulator.

Any reason why this wouldn't work? Let me know if you need to see the code.

**Comment by drewhannay** [comment]
Hm, I don't think this is something I've ever tried. Library modules ARE different than app modules, in that the library code and test code get packaged together in the same instrumentation APK (there's no separate standalone APK). I could imagine that this might cause problems for Test Butler, but I'm not sure exactly what the problem would be.

One workaround would be to have a sample app for your library and put the instrumentation tests there instead.

**Comment by zsperske** [comment]
Good idea! Would a sample project that demonstrates the library module issue be useful to you?

**Comment by drewhannay** [comment]
Sure, that can't hurt. We can investigate this a little more, but I suspect it might not be possible to fix. If it's not, we can at least document this in the README
